datameta/static/js/admin.js

```diff
@@ -956,21 +1075,155 @@ DataMeta.admin.updateGroup = function (group_id, name) {
             DataMeta.admin.reload();
         } else  if (response.status == "400") {
             response.json().then((json) => {
-                showAlert("group_alert", json[0].message);
+                showAlert("groups_alert", json[0].message);
             });
         } else if (response.status == "401") {
-            showAlert("group_alert", "You have to be logged in to perform this action.");
+            showAlert("groups_alert", "You have to be logged in to perform this action.");
         } else if (response.status == "403") {
-            showAlert("group_alert", "You do not have the rights to perform this action.");
+            showAlert("groups_alert", "You do not have the rights to perform this action.");
         } else {
-            showAlert("group_alert", "An unknown error occurred. Please try again later.");
+            showAlert("groups_alert", "An unknown error occurred. Please try again later.");
         }
     })
     .catch((error) => {
         console.log(error);
     });
 }
 
+// Adds a User with a specific ID to the service
+DataMeta.admin.addServiceUser = function(uuid) {
+    
+    // Get the new user id
+    var newUser = document.getElementById('user-id-input-' + uuid).value;
+
+    // Get the ids of the current users of this service
+    var userIds = [];
+    DataMeta.admin.services.forEach((service) => {
+        if(service.id.uuid == uuid) {
+            service.userIds.forEach((userId) => {
+                userIds.push(userId.uuid);
+            })
+        }
+    });
+    userIds.push(newUser);
+
+    // Perform an update of the service with the new list of user ids
+    DataMeta.admin.updateService(uuid, undefined, userIds);
+}
+
+// Enables editing of the Users of one Service
+DataMeta.admin.editServiceUsers = function(event, uuid) {
+    
+    // The button that triggered the function call
+    var button = event.srcElement;
+
+    // Enable Editing
+    button.parentNode.querySelectorAll('li').forEach((listElement) => {
+        listElement.innerHTML = '<input class="form-check-input me-2" type="checkbox" value="" id="flexCheckDefault">' + listElement.innerHTML;
```

Comment:
Let's have the users all checked initially so that a double click on the edit button does not accidentally delete all users
```suggestion
        listElement.innerHTML = '<input class="form-check-input me-2" type="checkbox" value="" id="flexCheckDefault" checked>' + listElement.innerHTML;
```